src/Analysis/Codelyzer.Analysis.Build/ProjectBuildHandler.cs

```diff
@@ -136,15 +140,16 @@ public ProjectBuildHandler(ILogger logger, Project project, AnalyzerConfiguratio
         }
         public async Task<ProjectBuildResult> Build()
         {
-            await SetCompilation();
+            var isSyntaxAnalysis = await SetCompilation();
```

Comment:
Name of the variable is confusing as the method returns false on falling back to syntax analysis. Or return true from method on syntax analysis

Reply:
+1 to Nagaraju's comment. My initial thought after seeing `SetCompilation()` return `false` is that the method failed to set a compilation, but since there is a fallback compilation, a compilation is always set.

Rather than renaming the variable or returning anything at all, what do you think about creating a new class property called something like `IsSemanticAnalysisAvailable`, `IsSyntaxAnalysisOnly`, or `FallbackCompilationUsed` and set it inside `SetCompilation()`?